packages/routing-utils/src/types.ts

```diff
@@ -21,7 +21,39 @@ export type HasField = Array<
     }
 >;
 
-export type Source = {
+type MiddlewareIndexed =
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware: number;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareKey?: never;
+    }
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware?: never;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareKey: string;
```

Comment:
I think `middlewareAt` would be a better name

Reply:
Usually `At` suffix means a timestamp such as `createdAt: number` or `updatedAt: number`.

I think `middlewareKey` or `middlewarePath` is better.

Reply:
`middlewarePath` seems good too